scipy/optimize/_gensa.py

```diff
@@ -0,0 +1,669 @@
+                    max_steps_not_exceeded = False
+                    break
+                # Need a re-annealing process? - Restarting main loop
+                if self._temperature < self.temperature_restart:
+                    break
+                # Starting Markov chain
+                self._run_markov_chain(i)
+
+                # Decision making for performing a local search
+                # based on the markov chain results
+                if not self._emin_unchanged and not self.pure_sa:
+                    temp = np.array(self._xmin)
+                    etemp = self._ls_energy(temp)
+                    temp = self._xbuffer
+                    if etemp < self._emin:
+                        self._xmin = np.array(temp)
+                        self._emin = np.array(etemp)
+                        self._index_no_emin_update = 0
+                if self._index_no_emin_update >= (
+                        self._index_tol_emin_update - 1) and not self.pure_sa:
+                    self._emin_markov = np.array(self._ls_energy(
+                        self._xmin_markov))
+                    self._index_no_emin_update = 0
+                    self._index_tol_emin_update = self._x.size
+                    if self._emin_markov < self._emin:
+                        self._xmin = np.array(self._xmin_markov)
+                        self._emin = np.array(self._emin_markov)
+                if self._check_stopping():
+                    self._stop_search()
+                    return 0
+        self._stop_search()
+        self._message = ["Number of iteration reached"]
+        return 0
+
+    def _check_stopping(self):
+        """Check if the search has to be stopped
+        """
+        if self.know_real:
+            if self._emin <= self.real_threshold:
+                self._message = ["Known value for minimum reached"]
+                return True
+        self._endtime = time.time()
+        delta = self._endtime - self._starttime
+        if delta >= self.maxtime:
+            self._message = ["Time limit reached"]
+            return True
+        if self._nbfuncall >= self.maxfuncall:
+            self._message = ["Number of function call reached"]
+            return True
+
+    def _stop_search(self):
+        """Record time stamp when stop searching
+        """
+        if self.pure_sa:
+            # In case of pure sa approach, doing ad LS at the end
+            temp = np.array(self._xmin)
+            etemp = self._ls_energy(temp)
+            temp = self._xbuffer
+            self._xmin = np.array(temp)
+            self._emin = np.array(etemp)
+        self._endtime = time.time()
+
+    def _coordin(self):
+        """Random generation of new coordinates
+        """
+        self._x = self._random_state.random_sample(
+            self._x.size) * self._xrange + self._lower
+
+    def _energy(self, x):
+        """Calling objective function and adding elasticity if needed
+        """
```

Comment:
elasticity?  would this be effective if the objective function grows faster than the elastic energy?  Why not implement the bounds constraint in the visiting distribution?